test/integration/api/forms/draft.js

```diff
@@ -123,9 +123,9 @@ describe('api: /projects/:id/forms (drafts)', () => {
         should.not.exist(body.enketoId);
       }));
 
-      it('should stop waiting for Enketo after 0.5 seconds @slow', testService(async (service) => {
+      it('should wait for Enketo only briefly @slow', testService(async (service) => {
         const asAlice = await service.login('alice');
-        global.enketo.wait = (f) => { setTimeout(f, 501); };
+        global.enketo.wait = (done) => { setTimeout(done, 600); };
```

Comment:
The change from 501 to 600 is meant to address this intermittent test failure: https://github.com/getodk/central-backend/issues/588#issuecomment-1722828303. Note that this test was not introduced in #989, but rather #782. I don't think I did anything in #989 that would have changed the POST/GET &hellip;/draft endpoints. However, if this test is failing, it certainly seems possible that new tests in #989 could fail, so I think it's worth addressing now.

Here's one possible explanation for the failure:

- A request to Enketo is sent. After 501 ms, the Enketo mock will respond.
- A moment later, the endpoint time-bounds the request. After 500 ms, the endpoint will stop waiting for the request.
- If "a moment later" is greater than 1 ms, then the test will fail.

Another possible explanation:

- `setTimeout()` doesn't run callbacks after an exact number of milliseconds, but after _at least_ the specified number of milliseconds. So the Enketo mock could respond after 502 ms instead of after 501 ms.
- Similarly, the endpoint could stop waiting after 502 ms instead of after 500 ms.
- In other words, both timers could exceed the number of milliseconds specified and maybe be processed together in a single batch. In that case, the order in which the callbacks are run might be indeterminate.

I'm not sure what the exact explanation is, but the fact that the test is failing intermittently doesn't make me think that there's anything wrong with the functionality here. I think we just need to provide a little more wiggle room for the test.